### PR TITLE
Expose more string primitives

### PR DESCRIPTION
**Pull request by david-christiansen**
This should make it reasonable to write some actual programs once IO support is merged.

### CODE REVIEW

File: examples/string.golden [hunk 1]
```diff
@@ -2,3 +2,40 @@
 "hello world" : String
 (true) : Bool
 (false) : Bool
+"cde" : (Maybe String)
```
Comment (langston-barrett) [inline examples/string.golden:5]: Shouldn't this have a `just` if it's a `Maybe String`?

File: src/Expander.hs [hunk 2]
```diff
@@ -417,6 +417,46 @@ initializeKernel = do
...
+        , ValueClosure $ HO $
+          \(ValueInteger (fromInteger -> start)) ->
+            ValueClosure $ HO $
+            \(ValueInteger (fromInteger -> len)) ->
+              ValueClosure $ HO $
+              \(ValueString str) ->
+                if | start < 0 || start       >= T.length str -> primitiveCtor "nothing" []
+                   | len   < 0 || start + len >  T.length str -> primitiveCtor "nothing" []
+                   | otherwise ->
+                     ValueString $ T.take len $ T.drop start str
```
Comment (langston-barrett) [inline src/Expander.hs:432]: I think this is missing a `just`?

File: examples/string.golden (hunk 1 above)
Comment (david-christiansen) [reply examples/string.golden:5]: Absolutely! Good catch!

File: src/Expander.hs (hunk 2 above)
Comment (david-christiansen) [reply src/Expander.hs:432]: Yep!

**Pull request merged**